charts/spire/charts/spiffe-csi-driver/values.yaml

```diff
@@ -37,6 +37,15 @@ healthChecks:
   ## @param healthChecks.port The healthcheck port for spiffe-csi-driver
   port: 9809
 
+## @param updateStrategy.type The update strategy to use to replace existing DaemonSet pods with new pods
+## @param updateStrategy.rollingUpdate.maxUnavailable Max unavailable pods during update
+## @param updateStrategy.rollingUpdate.maxSurge Max pods that simultatiously start the update
+updateStrategy:
+  type: RollingUpdate
```

Comment:
Is the plan to make this an enum of the available update strategies?

Reply:
personally I would never go with another strategy. It is in the values because I used the toYaml approach, maybe we should remove the option to change it 🤔

Reply:
the other option at the moment is 'OnDelete', which I think has some advantages too. Being able to keep things more stable by upgrading the daemonset but not the workload until your node is drained and in a downtime before deleting / upgrading the agent.

Reply:
@faisal-memon there is no concept of enums out of the box in yaml/helm. I don't have the time at this point in time to build our own validation using macros.

Reply:
@kfox1111 should we go with OnDelete as a more safe default?